spotbugs/src/main/java/edu/umd/cs/findbugs/cwe/WeaknessCatalog.java

```diff
@@ -0,0 +1,123 @@
+package edu.umd.cs.findbugs.cwe;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.UnsupportedEncodingException;
+import java.time.LocalDate;
+import java.util.HashMap;
+import java.util.Map;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonIOException;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonPrimitive;
+import com.google.gson.JsonSyntaxException;
+import com.google.gson.stream.JsonReader;
+
+/**
+ * The weakness catalog contains a number of weaknesses
+ *
+ * @author Jeremias Eppler
+ * @see Weakness
+ */
+public class WeaknessCatalog {
+    private String version;
+
+    private String name;
+    private LocalDate releaseDate;
+    private static final String FILE_NAME = "CWE_4.10.json";
+    private final Map<Integer, Weakness> weaknesses = new HashMap<>();
+
+    private static WeaknessCatalog INSTANCE;
+
+    private WeaknessCatalog() {
+    }
+
+    /**
+     * @return a weakness catalog instance
+     */
+    public synchronized static WeaknessCatalog getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new WeaknessCatalog();
+            loadFileAndInitialize();
+        }
+
+        return INSTANCE;
+    }
+
+    // loads the weaknesses from a JSON file
+    private static void loadFileAndInitialize() {
+        InputStream inputStream = WeaknessCatalog.class.getClassLoader().getResourceAsStream(FILE_NAME);
+        Gson gson = new Gson();
+
+        if (inputStream == null) {
+            throw new IllegalArgumentException("file not found! " + FILE_NAME);
+        }
+
+        try (JsonReader reader = new JsonReader(new InputStreamReader(inputStream, "UTF-8"))) {
+            JsonElement rootElement = JsonParser.parseReader(reader);
+            JsonPrimitive nameElement = rootElement.getAsJsonObject().get("name").getAsJsonPrimitive();
+            JsonPrimitive versionElement = rootElement.getAsJsonObject().get("version").getAsJsonPrimitive();
+            JsonPrimitive dateElement = rootElement.getAsJsonObject().get("date").getAsJsonPrimitive();
+
+            INSTANCE.version = versionElement.getAsString();
+            INSTANCE.name = nameElement.getAsString();
+            INSTANCE.releaseDate = LocalDate.parse(dateElement.getAsString());
+
+            JsonElement weaknessElements = rootElement.getAsJsonObject().get("weaknesses");
+
+            for (JsonElement weaknessElement : weaknessElements.getAsJsonArray()) {
+                Weakness weakness = gson.fromJson(weaknessElement, Weakness.class);
+
+                INSTANCE.weaknesses.put(Integer.valueOf(weakness.getCweId()), weakness);
+            }
+        } catch (UnsupportedEncodingException e) {
+            e.printStackTrace();
```

Comment:
@Jeeppler Please use logger here rather than print stack trace.

Reply:
@hazendaz did it.